### PR TITLE
[IMP] mail: simplify sub-channels ACLs

### PR DESCRIPTION
**Pull request by tsm-odoo**
Until now, sub-channels did not have their own `group_public_id`: ACLs checked the one of the parent instead. However, this can lead to complex queries.

This PR simplify sub-channels ACLs: `group_public_id` is computed based on the parent one so the behavior remains the same. However, since this information is available from the sub-channel itself, ACLs can be simplified.

upgrade: https://github.com/odoo/upgrade/pull/6778

### CODE REVIEW

**Review by seb-odoo — APPROVED** [review]
Do we currently have anything to prevent writing group of child to a different value than group of parent?

Other than that it's looking good.

@robodoo r+

**Comment by seb-odoo** [comment]
@robodoo r- delegate+

File: addons/mail/models/discuss/discuss_channel.py [hunk 1]
```diff
@@ -328,6 +326,15 @@ def write(self, vals):
                     channels=format_list(self.env, self.mapped("name")),
                 )
             )
+        if "group_public_id" in vals:
+            if failing_channels := self.filtered(lambda channel: channel.parent_channel_id):
+                raise UserError(
+                    self.env._(
+                        "Cannot change authorized group of: %(channels)s.",
+                        channels=", ".join(failing_channels.mapped("name")),
```
Comment (seb-odoo) [inline addons/mail/models/discuss/discuss_channel.py:334]: should use `format_list`

File: addons/mail/models/discuss/discuss_channel.py [hunk 2]
```diff
@@ -328,6 +326,15 @@ def write(self, vals):
                     channels=format_list(self.env, self.mapped("name")),
                 )
             )
+        if "group_public_id" in vals:
+            if failing_channels := self.filtered(lambda channel: channel.parent_channel_id):
+                raise UserError(
+                    self.env._(
+                        "Cannot change authorized group of: %(channels)s.",
```
Comment (seb-odoo) [inline addons/mail/models/discuss/discuss_channel.py:333]: To make the error more clear, add "of sub channel"?

**Review by seb-odoo — APPROVED** [review]
Otherwise ok for me, just need to get upgrade approved to more forward.

**Comment by tsm-odoo** [comment]
@robodoo r+

**Pull request closed without merging**